### ISSUE TITLE
Discord sends HTTP 411 (Length Required) when creating reaction events

### ISSUE THREAD

**Issue by fwcd** [nuclearace/SwiftDiscord#95]
When creating a reaction event, Discord responds with an error message indicating that the `Content-Length` header is missing from a `POST` request:

<details>
<summary><strong>Details & Log</strong></summary>

```
28.06.2020 23:16:20 [error] DiscordJSON: Invalid response code 411
28.06.2020 23:16:20 [error] DiscordJSON: Response: <!DOCTYPE html>
<html lang=en>
  <meta charset=utf-8>
  <meta name=viewport content="initial-scale=1, minimum-scale=1, width=device-width">
  <title>Error 411 (Length Required)!!1</title>
  <style>
    *{margin:0;padding:0}html,code{font:15px/22px arial,sans-serif}html{background:#fff;color:#222;padding:15px}body{margin:7% auto 0;max-width:390px;min-height:180px;padding:30px 0 15px}* > body{background:url(//www.google.com/images/errors/robot.png) 100% 5px no-repeat;padding-right:205px}p{margin:11px 0 22px;overflow:hidden}ins{color:#777;text-decoration:none}a img{border:0}@media screen and (max-width:772px){body{background:none;margin-top:0;max-width:none;padding-right:0}}#logo{background:url(//www.google.com/images/branding/googlelogo/1x/googlelogo_color_150x54dp.png) no-repeat;margin-left:-5px}@media only screen and (min-resolution:192dpi){#logo{background:url(//www.google.com/images/branding/googlelogo/2x/googlelogo_color_150x54dp.png) no-repeat 0% 0%/100% 100%;-moz-border-image:url(//www.google.com/images/branding/googlelogo/2x/googlelogo_color_150x54dp.png) 0}}@media only screen and (-webkit-min-device-pixel-ratio:2){#logo{background:url(//www.google.com/images/branding/googlelogo/2x/googlelogo_color_150x54dp.png) no-repeat;-webkit-background-size:100% 100%}}#logo{display:inline-block;height:54px;width:150px}
  </style>
  <a href=//www.google.com/><span id=logo aria-label=Google></span></a>
  <p><b>411.</b> <ins>That’s an error.</ins>
  <p>POST requests require a <code>Content-length</code> header.  <ins>That’s all we know.</ins>
```

</details>

Strangely, however, this issue only occurs when I run my bot in a Ubuntu 16.04 Docker container. On my Ubuntu 20.04 machine it seems to work fine. Any ideas why this particular header could be missing?

(I'm running a slightly modified version of the `vapor3` branch)

**Comment by fwcd** [comment]
I am still not sure whether the issue is on our side or Discord's API. `URLRequest` seems to generate `Content-Length` headers properly.

The fact that this issue came up out of the blue without changing the library makes me think that it might be a breaking API change (?). Maybe the change from `discordapp.com` to `discord.com` URLs has something to do with this?

Other Discord API wrappers seem to have faced the similar 411 errors, for varying reasons:
* https://github.com/discord-net/Discord.Net/issues/512 (closed)
* https://github.com/discord-net/Discord.Net/issues/1577 (recently opened)

The latter issue suggests that it might be a problem with newlines at the end of the bot token (or any other HTTP header value). I will further investigate this issue.

**Comment by TellowKrinkle** [comment]
BTW this was caused by this bug https://bugs.swift.org/browse/SR-12211 which was fixed here https://github.com/apple/swift-corelibs-foundation/pull/2735 so it should be fixed once Swift 5.3 comes out
